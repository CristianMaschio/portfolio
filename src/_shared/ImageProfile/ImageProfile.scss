@import "../../styles/colors";

.image-profile {
  border-radius: 50%;
}

.image-container {
  border-radius: 50%;
  border: 10px;
  background-image: $primary-right-linear-gradient;
  border-color: $dark-color;
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto;
  margin-top: 1em;
}
