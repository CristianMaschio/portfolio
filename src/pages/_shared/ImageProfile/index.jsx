import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'

import imageProfile from '../../../assets/images/imageProfile.jpg'

export default class ImageProfile extends PureComponent {
  render() {
    const { size } = this.props
    const cSize = size + size / 20

    return (
      <div className="image-container">
        <div
          className="image-background-animation"
          style={{ height: cSize, width: cSize }}
        />
        <img
          className="image-profile"
          alt="profile"
          src={imageProfile}
          style={{ height: size, width: size }}
        />
      </div>
    )
  }
}

ImageProfile.propTypes = {
  size: PropTypes.number.isRequired,
}

