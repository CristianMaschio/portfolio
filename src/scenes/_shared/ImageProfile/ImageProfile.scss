@import "../../../data/styles/colors";

@-webkit-keyframes rotating /* Safari and Chrome */ {
  from {
    -webkit-transform: rotate(0deg);
    -o-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  to {
    -webkit-transform: rotate(360deg);
    -o-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

@keyframes rotating {
  from {
    -ms-transform: rotate(0deg);
    -moz-transform: rotate(0deg);
    -webkit-transform: rotate(0deg);
    -o-transform: rotate(0deg);
    transform: rotate(0deg);
  }
  to {
    -ms-transform: rotate(360deg);
    -moz-transform: rotate(360deg);
    -webkit-transform: rotate(360deg);
    -o-transform: rotate(360deg);
    transform: rotate(360deg);
  }
}

.image-profile {
  position: absolute;
  border-radius: 50%;
}

.image-container {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: auto;
  margin-top: 1em;
}

.image-background-animation {
  border-radius: 50%;
  border: 10px;
  background-image: $primary-right-linear-gradient;
  border-color: $dark-color;
  -webkit-animation: rotating 4s linear infinite;
  -moz-animation: rotating 4s linear infinite;
  -ms-animation: rotating 4s linear infinite;
  -o-animation: rotating 4s linear infinite;
  animation: rotating 4s linear infinite;
}
