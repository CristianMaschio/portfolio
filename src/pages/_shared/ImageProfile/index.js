import React, { Component } from "react"
import PropTypes from "prop-types"

import "./ImageProfile.scss"
import imageProfile from "./imageProfile.jpg"

export default class ImageProfile extends Component {
  propTypes = {
    size: PropTypes.number.isRequired
  }

  render() {
    const { size } = this.props
    const cSize = size + size / 20

    return (
      <div className="image-container" style={{ height: cSize, width: cSize }}>
        <img
          className="image-profile"
          alt="profile"
          src={imageProfile}
          style={{ height: size, width: size }}
        />
      </div>
    )
  }
}
