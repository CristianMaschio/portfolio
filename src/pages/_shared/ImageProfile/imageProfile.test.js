import React from "react"
import renderer from "react-test-renderer"
import ImageProfile from "."

it("ImageProfile: render correctly", () => {
  const imageProfile = renderer.create(<ImageProfile size={20} />).toJSON()
  expect(imageProfile).toMatchSnapshot()
})