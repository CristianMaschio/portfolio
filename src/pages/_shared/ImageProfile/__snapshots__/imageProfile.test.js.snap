// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageProfile: render correctly 1`] = `
<div
  className="image-container"
>
  <div
    className="image-background-animation"
    style={
      Object {
        "height": 21,
        "width": 21,
      }
    }
  />
  <img
    alt="profile"
    className="image-profile"
    src="imageProfile.jpg"
    style={
      Object {
        "height": 20,
        "width": 20,
      }
    }
  />
</div>
`;
